### PR TITLE
Prevent JumpThreadingPass from messing with control points.

### PR DESCRIPTION
**Pull request by vext01**
Requires moving the -yk-patch-ctrl-point flag so it's accessible in two places.

### CODE REVIEW

File: llvm/lib/Transforms/Scalar/JumpThreading.cpp [hunk 1]
```diff
@@ -960,9 +962,25 @@ static bool hasAddressTakenAndUsed(BasicBlock *BB) {
   return !BA->use_empty();
 }
 
+// doesn't work need to move switch...
+//extern cl::opt<bool> YkPatchCtrlPoint;
```
Comment (ltratt) [inline llvm/lib/Transforms/Scalar/JumpThreading.cpp:966]: What's this bit for?

Comment (vext01) [reply llvm/lib/Transforms/Scalar/JumpThreading.cpp:966]: Oops. That's old junk left over.

Comment (ltratt) [reply llvm/lib/Transforms/Scalar/JumpThreading.cpp:966]: Please force push an update.

**Comment by vext01** [comment]
Force pushed the fix for that.

**Pull request merged**